clientcontroller/opstackl2/consumer.go

```diff
@@ -431,6 +433,48 @@ func ConvertProof(cmtProof cmtcrypto.Proof) Proof {
 	}
 }
 
+// GetBlockNumberByTimestamp returns the L2 block number for the given BTC staking activation timestamp.
+// It uses a binary search to find the block number.
+func (cc *OPStackL2ConsumerController) GetBlockNumberByTimestamp(ctx context.Context, targetTimestamp uint64) (uint64, error) {
+	// Check if the target timestamp is after the latest block
+	latestBlock, err := cc.opl2Client.HeaderByNumber(ctx, nil)
+	if err != nil {
+		return 0, err
+	}
+	if targetTimestamp > latestBlock.Time {
+		return 0, nil
+	}
+	// Check if the target timestamp is before the first block
+	firstBlock, err := cc.opl2Client.HeaderByNumber(ctx, big.NewInt(1))
+	if err != nil {
+		return 0, err
+	}
+	if targetTimestamp < firstBlock.Time {
```

Comment:
let's say block 0 is at t0 and block 1 at t1

if `t0 < t < t1`, the activated height should be block 1 but not block 0. so here should return 1 instead. wdyt?

Reply:
yes, it should return 1

Reply:
Would be good to clarify this in the docstring